systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java

```diff
@@ -174,6 +183,162 @@ void testJSONFormatLogging() {
+                        "appender.console.layout.type=PatternLayout\n" +
+                        "appender.console.layout.pattern=[%d] %-5p <%-12.12c{1}:%L> [%-12.12t] %m%n\n" +
+                        "rootLogger.level=DEBUG\n" +
+                        "rootLogger.appenderRefs=stdout\n" +
+                        "rootLogger.appenderRef.console.ref=STDOUT\n" +
+                        "rootLogger.additivity=false"))
+                .build();
+
+        kubeClient().getClient().configMaps().inNamespace(NAMESPACE).createOrReplace(configMapTo);
+        kubeClient().getClient().configMaps().inNamespace(NAMESPACE).createOrReplace(configMapUo);
+
+        ExternalLogging elTo = new ExternalLoggingBuilder()
+                .withName("external-configmap-to")
+                .build();
+
+        ExternalLogging elUo = new ExternalLoggingBuilder()
+                .withName("external-configmap-uo")
+                .build();
+
+        LOGGER.info("Setting log level of TO and UO to INFO - records should appear in log");
+        // change to external logging
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            k.getSpec().getEntityOperator().getTopicOperator().setLogging(elTo);
+            k.getSpec().getEntityOperator().getUserOperator().setLogging(elUo);
+        });
+
+        LOGGER.info("The EO shouldn't roll - verifying pod stability");
+        DeploymentUtils.waitForNoRollingUpdate(KafkaResources.entityOperatorDeploymentName(CLUSTER_NAME), eoPods);
+
+        LOGGER.info("Waiting for log4j2.properties will contain desired settings");
+        TestUtils.waitFor("Logger change", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> cmdKubeClient().execInPodContainer(eoPodName, "topic-operator", "cat", "/opt/topic-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=DEBUG")
+                        && cmdKubeClient().execInPodContainer(eoPodName, "user-operator", "cat", "/opt/user-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=DEBUG")
+                        && cmdKubeClient().execInPodContainer(eoPodName, "topic-operator", "cat", "/opt/topic-operator/custom-config/log4j2.properties").out().contains("monitorInterval=30")
+                        && cmdKubeClient().execInPodContainer(eoPodName, "user-operator", "cat", "/opt/user-operator/custom-config/log4j2.properties").out().contains("monitorInterval=30")
+        );
+
+        LOGGER.info("Waiting {}ms for INFO log will appear", RECONCILIATION_INTERVAL * 2);
+        // wait some time if TO and UO will log something
+        Thread.sleep(RECONCILIATION_INTERVAL * 2);
+
+        LOGGER.info("Asserting if log will contain some records");
+        assertThat(StUtils.getLogFromPodByTime(eoPodName, "user-operator", "5m"), is(not(emptyString())));
+        assertThat(StUtils.getLogFromPodByTime(eoPodName, "topic-operator", "5m"), is(not(emptyString())));
```

Comment:
I wonder if we should follow with another part where you check that changes to external logging config map were applied as well.